app/src/main/java/dev/nadeldrucker/trafficswipe/fragments/StartFragment.java

```diff
@@ -107,11 +122,19 @@ private void setSoftKeyboardState(boolean isShown, View view){
 
     /**
      * Displays the departure table using the specified query request.
-     * @param query name to query
+     * Or show an error if result is null
+     * @param result Abbreviation to query
      */
-    protected void showDepartureTable(String query){
-        new ViewModelProvider(Objects.requireNonNull(getActivity())).get(DeparturesViewModel.class).getUserStationName().postValue(query);
-        Navigation.findNavController(Objects.requireNonNull(getView())).navigate(R.id.action_startFragment_to_resultFragment);
+    private void handleResult(Abbreviation... result) {
```

Comment:
gets called everytime the abbreviations change -> If you go back to the start you are instantly navigated back to the Results fragment.